[Synthetics] Default Rule creation

justinkambic opened the pull request:
## Summary

Resolves #241396.

Primary changes to how we create default Synthetics rules:

- Navigation-based events will no longer trigger the creation of the rules
- Creating a monitor will create the rules
- Editing a monitor will create the rules

## Testing this PR

1. In a clean deployment/space, open two Kibana tabs, one for o11y rules page, the other some page besides Synthetics.
1. Check the rules page, there should be no default Synthetics rules.
1. In your second tab, navigate to Synthetics.
1. Check the rules page again and ensure there are no Synthetics default rules.
1. Create a Synthetics monitor.
1. Go to the Settings tab in Synthetics and verify that there are no connectors set up (see Fig. 1 below).
1. Check the rules page again, and see that there are monitor status and TLS default alerts now. Delete the rules.
1. In Synthetics, navigate to the overview page and the management page within the app.
1. Check the rules page again, see that there are still no new rules.
1. Edit your monitor (change the name or something).
1. After successfully saving the edits to your monitor, check the rules page again. See that there are now default monitor status and TLS alerts once more.

### Figure 1, empty default connectors

<img width="831" height="103" alt="image" src="https://github.com/user-attachments/assets/08b7339d-d21f-4e84-af66-4a6717c78f41" />

justinkambic committed:
Do not create monitors during regular navigation.

justinkambic committed:
Fix edit default rules popover.

justinkambic committed:
Create rules even when connectors are not set up.

justinkambic committed:
Fix an API test that asserts that the rules do not get created when connectors are undefined.

elasticmachine commented:
Pinging @elastic/actionable-obs-team (Team:actionable-obs)

elasticmachine commented:
Pinging @elastic/obs-ux-management-team (Team:obs-ux-management)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/369741)
* Commit: 6ded1f8c739a050702b3639ad2ad0f2f7e841ac5

### Failed CI Steps
* [Synthetics @elastic/synthetics Tests](https://buildkite.com/elastic/kibana-pull-request/builds/369741#019af029-8df8-4c57-abad-58c1e540fdb8)
* [Synthetics @elastic/synthetics Tests](https://buildkite.com/elastic/kibana-pull-request/builds/369741#019af05a-b72d-4f2c-b07d-703429927a4a)

cc @justinkambic

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369741","number":369741,"commit":"6ded1f8c739a050702b3639ad2ad0f2f7e841ac5"}],"number":369741}
buildkite-pr-comment-->

justinkambic committed:
Update synthetics test.

justinkambic committed:
Add unit tests.

justinkambic committed:
Add unit tests.

justinkambic committed:
PR feedback.

justinkambic committed:
Add tests for disabled state.

justinkambic committed:
PR feedback.

justinkambic committed:
More tests.

justinkambic committed:
PR feedback.

justinkambic committed:
Merge remote-tracking branch 'upstream/main' into 241396/create-default-rules-without-connectors-and-do-not-create-default-rules-during-navigation

justinkambic committed:
Merge branch 'main' of github.com:elastic/kibana into 241396/create-default-rules-without-connectors-and-do-not-create-default-rules-during-navigation

justinkambic committed:
Merge remote-tracking branch 'upstream/main' into 241396/create-default-rules-without-connectors-and-do-not-create-default-rules-during-navigation

justinkambic committed:
Merge remote-tracking branch 'upstream/main' into 241396/create-default-rules-without-connectors-and-do-not-create-default-rules-during-navigation

justinkambic committed:
Enhance alert toggle button tooltip logic and update synthetics rules hook

- Updated the tooltip content logic in the ToggleAlertFlyoutButton component to handle cases where the status rule does not exist, providing a clearer user experience.
- Refactored the useSyntheticsRules hook to include additional imports and improve functionality, enhancing the management of alert rules and settings.

justinkambic committed:
Merge remote-tracking branch 'upstream/main' into 241396/create-default-rules-without-connectors-and-do-not-create-default-rules-during-navigation

justinkambic committed:
Merge remote-tracking branch 'upstream/main' into 241396/create-default-rules-without-connectors-and-do-not-create-default-rules-during-navigation

justinkambic committed:
Update ToggleAlertFlyoutButton test to reflect correct tooltip behavior when user has permissions and status rule exists.

justinkambic committed:
Add test per PR feedback.

justinkambic committed:
Update test name per PR feedback.

justinkambic commented on x-pack/solutions/observability/plugins/synthetics/server/routes/default_alerts/default_alert_service.ts:
```diff
@@ -48,13 +47,6 @@ export class DefaultRuleService {
 
   async setupDefaultRules() {
     this.settings = await this.getSettings();
-    if (isEmpty(this.settings?.defaultConnectors)) {
```
We are now creating the rules regardless of whether the connectors are configured or not.

fkanout reviewed (commented):
Tested locally. 
- Rules created on edit (if they are no rules)
- Rules created on creating the monitor (even without a connector)
- NO Rules created on edit (when `SYNTHETICS_DEFAULT_ALERT ` rules are created)

All the above match the expected behavior 👍🏻 

⚠️ However, before this PR, default rules were only created via the "getting started" flow and tagged with `SYNTHETICS_DEFAULT_ALERT` and potentially deleted by the user. Existing users who created monitors through the regular UI/API have no rules with this tag. After this PR, editing any monitor triggers `initDefaultAlerts()`, which doesn't find the tag, so it creates duplicate rules (see screen recording)

https://github.com/user-attachments/assets/3ecabf6d-cd31-4a4d-82ec-abaae880dfe1

justinkambic committed:
Merge branch 'main' into 241396/create-default-rules-without-connectors-and-do-not-create-default-rules-during-navigation

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/379230)
* Commit: ba70cfaf87b5c181f7723f57228166c2c5338a6d

### Failed CI Steps
* [FTR Configs #17](https://buildkite.com/elastic/kibana-pull-request/builds/379230#019b9e96-3670-44ca-bb41-501f4f11a5ad)
* [Scout: [ platform / streams_app ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/379230#019b9e9d-29d8-4dac-bb01-582749f6bade)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/379230#019b9e96-3670-44ca-bb41-501f4f11a5ad) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/379230/jobs/019b9e96-3670-44ca-bb41-501f4f11a5ad/artifacts/019b9ea2-f7a6-47f9-9fcd-dce5a8954e7f) FTR Configs #<span></span>17 / alerting api integration security and spaces enabled - Group 2 Connectors webhook action OAuth2 client credentials should refresh the token once the previous one has expired

### History
* :yellow_heart: [Build #378112](https://buildkite.com/elastic/kibana-pull-request/builds/378112) was flaky 4c55aa7d5e9da1b4f108b6c7d82d98047afac4b5
* :green_heart: [Build #378077](https://buildkite.com/elastic/kibana-pull-request/builds/378077) succeeded 681caa1f42548a3c56d28b9ba21aa4c4a0e0bef2
* :broken_heart: [Build #378005](https://buildkite.com/elastic/kibana-pull-request/builds/378005) failed 61f0939a095d0b220751170f06c59830aaf10130
* :broken_heart: [Build #376283](https://buildkite.com/elastic/kibana-pull-request/builds/376283) failed 5ed7c28a7a6fee22ce7e0de48ae6ac63480cc019
* :broken_heart: [Build #375238](https://buildkite.com/elastic/kibana-pull-request/builds/375238) failed 8b9ea532445c7987063d5f7d2d6efdf46d18f2ab
* :broken_heart: [Build #373484](https://buildkite.com/elastic/kibana-pull-request/builds/373484) failed eed8c501309cbf639f41dfbdbaf75bf25ff8519e

cc @justinkambic

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/379230","number":379230,"commit":"ba70cfaf87b5c181f7723f57228166c2c5338a6d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/378112","number":378112,"commit":"4c55aa7d5e9da1b4f108b6c7d82d98047afac4b5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/378077","number":378077,"commit":"681caa1f42548a3c56d28b9ba21aa4c4a0e0bef2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/378005","number":378005,"commit":"61f0939a095d0b220751170f06c59830aaf10130"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376283","number":376283,"commit":"5ed7c28a7a6fee22ce7e0de48ae6ac63480cc019"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375238","number":375238,"commit":"8b9ea532445c7987063d5f7d2d6efdf46d18f2ab"}],"number":379230}
buildkite-pr-comment-->
